### DISCUSSION TITLE
Android Notification - Popup with a Silent sound

### DISCUSSION THREAD

**Discussion by munkii** [shinyorg/shiny#1067] [Q&A] [answered]
We want to default our Push Notifcations for Android to display as a "Pop-Up". The only way I can see that that is possible is by using a channel and setting the Channel to be an importance of High.

Whilst that does mean we get a Pop-Up it also means by default we get a Sound and for some annoying reason that sound is on repeat until the notifcation is tapped. Imagine how annoying that is as a user! :-)

As a user i can change that sound to be "Silent" but we don't want to ask users to have to do that. How do we code the app so that the notifcation is still a Pop-Up but with no Sound?

Can I set the CustomSoundPath of the Shiny.Notifications.Channel to a non exisitent file to essentially silence it?

**Comment by aritchie** [comment, marked as answer]
You won't be able to set it to a non-existing file as you'll get an error when it tries to use the sound within the notification.  

I've been thinking of ideas to make the sounds more customizable for channels.  I'll see what I can come up with for a future release.

**Reply by aritchie** [reply]
I'll put this in 2.6 - https://github.com/shinyorg/shiny/commit/c6103914d33c13c6621d45b137ff7a94e5bfdb54

**Comment by munkii** [comment]
Why is the PopUp style event so linked to it making a sound and why oh why does the sound repeat? I realise these are most likely Android issues rather than any issue with Shiny but I don't know where else to ask.

Are people that have successully implemeneted Push notifications with Android sworn to secrecy? :-)

**Reply by aritchie** [reply]
I think it may actually be the channel importance level.  I'm actually not sure.  

Is this a specific flavour of android perhaps?

**Reply by munkii** [reply]
Channel Importance Level is definetly key and in my experience withiout setting that to High 

`Importance = Shiny.Notifications.ChannelImportance.High,`

You do not get the notifcation "popping up" however tied to that settings is a sound which is on repeat. Our users were understandably unimpressed in facty I woke one of them up! :-)

You can, as a user disentangle "Popping up" from the infernalnoise by setting the sound to "Silent" (note not the notifcation to silent). I can't believe this is so painful.

**Reply by aritchie** [reply]
Android is always the pain point for this library - ALWAYS!  

Please try the latest v2.6 with the channel.Sound = ChannelSound.None.  Let me know how that works for you.

**Reply by munkii** [reply]
`channel.Sound = ChannelSound.None`

Worked perfectly thanks. The only problem is as you probably know once a channel has been created (existing installs) you can't do much to them other than change their name, but for new installs this works. I'll look at what to do for existing installs

So far I have only tried it on an Emulator (I'll try a device soon)
